[Docs] Japanese translation of docs/feature_leader_key.md (#8950)
qmk/qmk_firmware: docs/ja/feature_leader_key.md

# リーダーキー: 新しい種類のモディファイア

<!---
  original document: 0.8.134:docs/feature_leader_key.md
  git diff 0.8.134 HEAD -- docs/feature_leader_key.md | cat
-->

もしあなたが Vim を使ったことがある場合、リーダーキーは何であるかを知っています。そうでなければ、素晴らしい概念を発見しようとしています。:) 例えば、Alt+Shift+W を押す(3つのキーを同時に押す)代わりに、キーの_シーケンス_を押すとどうなりますか？つまり、特別なモディファイア (リーダーキー)を押して、続けて W と C を押します (単純にキーを高速に繋げます)。そして何かが起こります。

それが `KC_LEAD` の機能です。以下は例です:

1. リーダーキーとして使いたいキーボードのキーを選択します。それをキーコード `KC_LEAD` に割り当てます。このキーはこのためだけの専用です -- 単一のアクションキーで、他には使うことができません。

> 2. `config.h` 内に `#define LEADER_TIMEOUT 300` の行を含めます。これは `KC_LEAD` キーのタイムアウトを設定します。具体的には、`KC_LEAD` キーを押した場合、リーダーキーのシーケンスを完了するためには一定の時間しかありません。ここでの `300` はそれを300msに設定します。この値を増やして、シーケンスがヒットする時間を増やすことができます。ただし、このタイムアウト中に押された全てのキーは全て途中で捉えられ、送信されません。そのためこの値を小さくしておきたいかもしれません。.

Author: 訳例ありがとうございます。

>    * デフォルトでは、このタイムアウトは、`KC_LEAD` を押した後にシーケンス全体が完了するまでどれだけ掛かるかの時間です。これは一部の人にとっては非常に低いかもしれません。そのため、このタイムアウトを増やしたほうが良いかもしれません。オプションで、`LEADER_PER_KEY_TIMING` オプションを有効にしたほうが良いかもしれません。これは各キーがタップされた後でタイムアウトを再設定します。これにより、ここでは低い値を設定しておき、より長いシーケンスを使うことができます。このオプションを有効にするには、`config.h` に `#define LEADER_PER_KEY_TIMING` を追加します。

Author: 訳例ありがとうございます。すごく読みやすくなりました！

3. `matrix_scan_user` 関数の中で、以下のようなものを追加します:

```c
LEADER_EXTERNS();

void matrix_scan_user(void) {
  LEADER_DICTIONARY() {
    leading = false;
    leader_end();

    SEQ_ONE_KEY(KC_F) {
      // マクロ内でできること
      SEND_STRING("QMK is awesome.");
    }
    SEQ_TWO_KEYS(KC_D, KC_D) {
      SEND_STRING(SS_LCTL("a") SS_LCTL("c"));
    }
    SEQ_THREE_KEYS(KC_D, KC_D, KC_S) {
      SEND_STRING("https://start.duckduckgo.com\n");
    }
    SEQ_TWO_KEYS(KC_A, KC_S) {
      register_code(KC_LGUI);
      register_code(KC_S);
      unregister_code(KC_S);
      unregister_code(KC_LGUI);
    }
  }
}
```

ご覧のとおり、幾つかの機能があります。`SEQ_ONE_KEY` を単一キーシーケンス (リーダーの後に1つのキーのみ)に使い、より長いシーケンスについては `SEQ_TWO_KEYS`、`SEQ_THREE_KEYS` から `SEQ_FIVE_KEYS` を使うことができます。

これらはそれぞれ1つ以上のキーコードを引数として受け付けます。これは重要な点です: **キーボードの任意のレイヤー**のキーコードを使うことができます。当たり前ですが、リーダーマクロが発動するにはそのレイヤーがアクティブである必要があります

## `rules.mk` にリーダーキーサポートを追加

リーダーキーのサポートを追加するには、単純にキーマップの `rules.mk` に1行を追加します:

```make
LEADER_ENABLE = yes
```

## リーダーキーのキーごとのタイミング

> 長いリーダーキー文字列のために非常に長いタイムアウト時間に頼ったり、200wpm のタイピングスキルが無い場合、キーを押すごとに入力を完了するまでの時間を増やす機能を使用することができます。これは、リーダーキーを使ってタップダンスを再現する場合に非常に役立ちます (C, C, C のような同じキーを複数回タップする場合)。

Author: ご指摘ありがとうございます。

> 長いリーダーキー文字列のために長いタイムアウト時間を設定したくない場合や、200wpm のタイピングスキルが無い場合、キーを押すごとに入力を完了するまでの時間を増やす機能を使用することができます。これは、リーダーキーを使ってタップダンスを再現する場合に非常に役立ちます (C, C, C のような同じキーを複数回タップする場合)。

Reviewer: 「Rather 〜 for long leader key strings」の中に「したくない」という否定を表す単語が無いので、次の訳はどうでしょうか。
```suggestion
長いリーダーキー文字列のために非常に長いタイムアウト時間に頼ったり、200wpm のタイピングスキルが無い場合、キーを押すごとに入力を完了するまでの時間を増やす機能を使用することができます。これは、リーダーキーを使ってタップダンスを再現する場合に非常に役立ちます (C, C, C のような同じキーを複数回タップする場合)。
```

Author: ご指摘ありがとうございます。元の訳文の回りくどさがなくなった気がします。

Reviewer 2: Rather than relying on an incredibly high timeout （略）we can ...は、「非常に長いタイムアウト時間に頼りたくない場合は代わりに...」という意味なので、「長いリーダーキー文字列のために非常に長いタイムアウト時間に頼ったり（略）キーを押すごとに入力を完了するまでの時間を増やす機能を使用することができます。」では、前半部の代替手段として後半部があるというのが伝わりにくいと考えています。
（また「たり」が1つしかないのも日本語の一般的な用法から外れます）
変更前の和訳は、元の文におけるan incredibly high timeout for long leader key stringsとthose of us without 200wpm typing skillsの並列を直訳しても日本語としては意味が非常に通じにくいので、多少意訳になっても日本語として意味が伝わりやすい文にすべきとの考えで作成しました。

Reviewer: > Rather than relying on an incredibly high timeout （略）we can ...は、「非常に長いタイムアウト時間に頼りたくない場合は代わりに...」という意味なので、「長いリーダーキー文字列のために非常に長いタイムアウト時間に頼ったり（略）キーを押すごとに入力を完了するまでの時間を増やす機能を使用することができます。」では、前半部の代替手段として後半部があるというのが伝わりにくいと考えています。

この点はご指摘通りで、私も悩んだ末に強引に上記のとおり訳したところです。それでも、依然として後半部の存在意義が伝わりにくいので、あまり役に立たない提案をしたことになってしまいました。

原文通りに訳し、かつ、日本語として通じる文章にするのは相当に難しそうなので、意訳になるのは仕方ないと考えて、

「長いリーダーキー文字列のために非常に長いタイムアウト時間に頼る、または200wpm のタイピングスキルが無い場合、代わりにキーを押すごとに入力を完了するまでの時間を増やす機能を使うことができます。」

という具合に、「または」と「代わりに」の2単語を追加するのはどうでしょうか？

Reviewer 2: 並列なので「頼る」＋「～ない場合」とするより、「頼りたくない場合」＋「～ない場合」と揃えた方が日本語としては自然な感じになると思います。

「長いリーダーキー文字列のために非常に長いタイムアウト時間に頼りたくない場合、または200wpm のタイピングスキルが無い場合、代わりにキーを押すごとに入力を完了するまでの時間を増やす機能を使うことができます。」

Reviewer: 確かに、語尾を揃えるほうが自然な感じになりますね。
「without 200wpm typing skills」は「200wpm のタイピングスキルが無い場合」という訳になりますから、「Rather than 〜 strings」に否定系の単語はないですが「スキルが無い場合」に合わせる形で「頼りたくない場合」と訳すのは妥当だと思います。

> 信じられないほど長いリーダーキー文字列についての高いタイムアウトや、200wpm のタイピングスキルの無い私たちに依存するのではなく、押された各キーがストロークを完了するのにより多くの時間を提供できるようにするために各キーごとにタイミングを有効にすることができます。これは、タップダンスのリーダーキーエミュレーションで非常に役立ちます (C, C, C のような同じキーの複数のタップ)。

Reviewer 2: たいぶ意訳になってしまいましたが問題ないでしょうか？
```suggestion
長いリーダーキー文字列のために長いタイムアウト時間を設定したくない場合や、200wpm のタイピングスキルが無い場合、キーを押すごとに入力を完了するまでの時間を増やす機能を使用することができます。これは、リーダーキーを使ってタップダンスを再現する場合に非常に役立ちます (C, C, C のような同じキーを複数回タップする場合)。
```

これを有効にするには、以下を `config.h` に配置します:
```c
#define LEADER_PER_KEY_TIMING
```

> この後、`LEADER_TIMEOUT` を300ms未満に下げることをお勧めします。

Reviewer 3: ```suggestion
この後、`LEADER_TIMEOUT` を 300ms 未満に下げることをお勧めします。
```

```c
#define LEADER_TIMEOUT 250
```

> これで、1000ms のリーダーキータイムアウト無しに行うことが可能に思われます。

Reviewer 2: ```suggestion
これで、リーダーキーのタイムアウト時間を 1000ms に設定することなく以下のようなことが可能になると思われます。
```

```c
SEQ_THREE_KEYS(KC_C, KC_C, KC_C) {
  SEND_STRING("Per key timing is great!!!");
}
```

## 厳密なキー処理

> デフォルトでは、リーダーキー機能は、リーダーシーケンスの確認時に [`モッドタップ`](ja/mod_tap.md) および [`レイヤータップ`](ja/feature_layers.md#switching-and-toggling-layers) 機能からのキーコードをフィルターします。つまり、`LT(3, KC_A)` を使っている場合、シーケンスの `KC_A` として取り出され、`LT(3, KC_A)` ではなく、新しいユーザにとってより期待される動作を提供します。

Reviewer: [英語の長文内で、「rather than」が文頭に来ている文章があり... - Yahoo!知恵袋](https://detail.chiebukuro.yahoo.co.jp/qa/question_detail/q10212674540) で紹介されている「rather than B A」のパターンだと思いますので、次の通り入れ替えたほうが良いと思います。
```suggestion
デフォルトでは、リーダーキー機能は、リーダーシーケンスの確認時に [`モッドタップ`](ja/mod_tap.md) および [`レイヤータップ`](ja/feature_layers.md#switching-and-toggling-layers) 機能からのキーコードをフィルターします。つまり、`LT(3, KC_A)` を使っている場合、`LT(3, KC_A)` ではなくシーケンスの `KC_A` として取り出され、新しいユーザにとってより期待される動作を提供します。
```

Author: ご指摘ありがとうございます。
訳例をそのまま頂きました。

> ほとんどの場合これで問題ありませんが、シーケンスでキーコード全体(例えば、上の例での `LT(3, KC_A)`) を指定したい場合は、`config.h` ファイルに `#define LEADER_KEY_STRICT_KEY_PROCESSING` を追加することで行えます。これでフィルタリングが無効になり、キーコード全体を指定する必要があります。

Reviewer 2: ```suggestion
ほとんどの場合これで問題ありませんが、シーケンスでキーコード全体(例えば、上の例での `LT(3, KC_A)`) を指定したい場合は、`config.h` ファイルに `#define LEADER_KEY_STRICT_KEY_PROCESSING` を追加することこのような機能を有効にすることができます。これでフィルタリングが無効になり、キーコード全体を指定する必要があります。
```

## カスタマイズ

> リーダーキー機能には、リーダーキー機能の動作にいくらかのカスタマイズを追加する方法があります。プロセスの特定の部分で呼び出すことができる2つの関数があります。つまり、`leader_start()` と `leader_end()` です。

Reviewer 2: ```suggestion
リーダーキー機能には、リーダーキー機能の動作にいくらかのカスタマイズを追加する方法があります。リーダーキー機能のプロセスの特定の部分で呼び出すことができる2つの関数、`leader_start()` と `leader_end()` です。
```

> `KC_LEAD` キーがタップされた時に `leader_start()` 関数が呼ばれ、リーダーシーケンスが完了するか、リーダータイムアウトがヒットした時に `leader_end()` 関数が呼ばれます。

Reviewer 2: ```suggestion
`KC_LEAD` キーがタップされた時に `leader_start()` 関数が呼ばれ、リーダーシーケンスが完了するか、リーダータイムアウトの時間に達した時に `leader_end()` 関数が呼ばれます。
```

> リーダーシーケンスにフィードバックを追加するために、これらの関数をコード (通常 は`keymap.c`)に追加することができます (ビープまたは音楽を再生するなど)。

Reviewer 3: ```suggestion
リーダーシーケンスにフィードバック(ビープまたは音楽を再生するなど)を追加するために、これらの関数をコード (通常 は`keymap.c`)に追加することができます。
```

```c
void leader_start(void) {
  // シーケンスの開始
}

void leader_end(void) {

>   // シーケンスの終了 (成功/失敗の検知無し)

Reviewer 3: ここは / の前後で区切られます。しかしながら、Exampleを見る限りsuccess/failuer detectionが正しそうなんですが…。
```suggestion
  // シーケンスの終了 (成功しない/失敗を検知)
```

Author: ご指摘ありがとうございます。

}
```

### 例

> この例では、リーダーシーケンスを開始するために `KC_LEAD` を押すとマリオの "One Up" 音が再生され、正常に完了した場合は "All Star" を再生され、失敗した場合は "Rick Roll" を再生します。

Reviewer 3: 「再生される」と「再生する」が混じっているので揃えたほうが良いです。
```suggestion
この例では、リーダーシーケンスを開始するために `KC_LEAD` を押すとマリオの "One Up" 音が再生され、正常に完了した場合は "All Star" が再生され、失敗した場合は "Rick Roll" が再生されます。
```

Author: 訳例ありがとうございます。
